README.md

```diff
@@ -7,208 +7,194 @@
 
-### Special features
-- Allow you to use Go libraries dynamically (only on linux)
-- Support calling Go's method using Goby directly
-- Built in multi-threaded server and DB library
+## Language
 
-### Normal language features
+Perhaps Goby should be far easier for Rubyists to comprehend. You can use Ruby's syntax highlighting for Goby as well😀
+ 
 - Everything is object
 - Object and Class
     - Top level main object
     - Constructor
-    - Support class methods
-    - Support inheritance
-    - Support instance variable
-    - Support `self`
+    - Class/instance method
+    - Class
+        - inheritance
+    - `self`
 - Module & Namespace
-- Variables
-    - Constant
+    - `::` for namespace
+- Variable: starts with lowercase letter like 'var`
     - Local variable
     - Instance variable
-- Method
-    - Support evaluation with arguments
-    - Support evaluation without arguments
-    - Support evaluation with block (closure)
-- BuiltIn Data Types (All of them are classes 😀)
-    - Class
-    - Integer
-    - String
-    - Boolean
-    - nil
-    - Hash (with built in `to_json` method)
-    - Array
+- Constant
+    - Starts with uppercase like `Var` or `VAR`
+    - global if defined on top-level 
+    - **not reentrant**
+    - (special variables with `$` are unsupported)
+- Methods 
+    - Evaluation with arguments
+    - Evaluation without arguments
+    - Evaluation with block (closure)
+- Block
+    - `do` - `end`
 - Flow control
-    - If statement
-    - while statement
-- Import other files
-    - require_relative
-    - require (only for standard libraries now)
-- Standard Libraries (all of them are at very early stage)
-    - `URI`
-    - `Net::HTTP`
-    - `Net::SimpleServer` (This is very cool and quite performante, check the [sample](https://github.com/goby-lang/goby/blob/master/samples/server.gb))
-    - `File`
-    - `DB`
+    - `if`, `else`, `elsif`
+    - `while`
 - IO
-    - `puts`
+    - `#puts`
     - `ARGV`
-- REPL (run `goby -i`)
-- Thread (this should work but the implementation is quite naive and will be refined in the future)
-    - Support `thread` method to create a new thread (like `goroutine`)
-    - Has `Channel` class for passing objects between threads (like `chan` in Go)
+- Import files
+    - `require` (Just for standard libraries by now)
+    - `require_relative`
+- Thread (not a class!)
+    - Goroutine-based `thread` method to create a new thread
+    - Works with `Channel` class for passing objects between threads, like `chan` in Go
     - See this sample: [One thousand threads](https://github.com/goby-lang/goby/blob/master/samples/one_thousand_threads.gb)
 
-    
-**(You can open an issue for any feature request)**
+## Native class
+ 
+Written in Go.
+
+- Class
+- Integer
+- String
+- Boolean
+- Null (nil)
+- Hash (with `#to_json` method)
+- Array (with `#each`)
+- Range (with `#each`)
+- URI
+- Channel
```

Comment:
There's a missing class: `GoObject`. Which stores any type of data that can't be transform to Goby's type directly. And we can use `GoObject#go_func` method to call the Go object's function.